index/manifest.go

```diff
@@ -0,0 +1,33 @@
+package index
+
+import (
+	"github.com/filecoin-project/dagstore/shard"
+	"github.com/ipfs/go-cid"
+)
+
+// ManifestKey identifies a manifest. It is a triple that can act as a composite
+// key, comprising the shard key and generation metadata.
+type ManifestKey struct {
+	Shard      shard.Key
+	GenRule    string
+	GenVersion uint
+}
+
+// Manifest are sets of CIDs with no offset indication.
+type Manifest interface {
```

Comment:
We've captured the concept of the Manifest in the design docs.
I suggest we don't add interfaces / implement the Manifest until we need it in practice.

Reply:
I would prefer to draw the shape of the system entirely even if we don't implement them. We can comment out the embedding of the Manifest interface in the `index.Repo`.